src/features/cluster/modals/NewInstanceModal.tsx

```diff
@@ -165,7 +157,7 @@ function NewInstanceModal({ clusterId }: { clusterId: string }) {
 												<SelectGroup>
 													{instanceTypes?.map((type) => (
 														<SelectItem key={type.id} value={type.id}>
-															{renderInstanceTypeOption(type.id)}
+															{renderInstanceTypeOption(type.id as InstanceTypes)}
```

Comment:
could this type assertion be avoided somehow, especially since this appears to be dynamic data, since it doesn't really do anything for the app? could try to ensure the data is appropriately validated and typed? food for thought later. :)

Reply:
I'll if I can, I was having a ton of trouble with this 😓. Would love to pair on this!